client_test.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+
+func TestNewClient(t *testing.T) {
+	Convey("Given acceptable input values", t, func(){
+		c, err := NewClient("https://localhost:4646", "testdata/testCertFile", false)
+		Convey("When a newClient is created", func() {
+			Convey("Then a client is returned with no errors", func(){
+				So(c, ShouldNotBeNil)
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+}
+
+func TestNewClientFails(t *testing.T) {
+	Convey("Given invalid input values", t, func(){
+		c, err := NewClient("https://localhost:4646", "", false)
+		Convey("When a new Client is created", func(){
+			Convey("Then no client is returned with errors", func(){
+				So(err, ShouldNotBeNil)
+				So(c, ShouldBeNil)
+				So(err.Error(), ShouldStartWith, "invalid configuration with https")
+			})
+		})
+	})
+}
+
+func TestNewClientHTTP(t *testing.T) {
+	Convey("Given valid input values", t, func(){
```

Comment:
This test looks like same as the previous one but the above fails and this one is successful. Have I misread it?